core/src/main/java/net/opentsdb/aura/metrics/core/data/FixedSizedMemoryBlockStream.java

```diff
@@ -0,0 +1,236 @@
+/*
+ * This file is part of OpenTSDB.
+ * Copyright (C) 2021  Yahoo.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.opentsdb.aura.metrics.core.data;
+
+import net.opentsdb.collections.DirectArray;
+import net.opentsdb.collections.DirectByteArray;
+import net.opentsdb.collections.DirectLongArray;
+
+/**
+ * It's a chain of fixed sized block of memory. The address to the next block is stored at the
+ * initial 8 byte of the current memory block. It takes additional 10 bytes of the first block to
+ * store two additional header fields. 8 bytes for the address of the last block and 2 bytes for the
+ * bit index in the last block.
+ *
+ * <p>The first block of the segment is shared across header and data. The {@link #header} block
+ * uses a {@link DirectByteArray} and should end at a byte boundary.The {@link #dataBlock} should
+ * start at a byte boundary.
+ *
+ * @see DirectByteArray
+ * @see DirectLongArray
+ */
+public abstract class FixedSizedMemoryBlockStream implements MemoryBlock, BitMap {
```

Comment:
Assuming it's all copy-pasta into this class.

Reply:
yes. This a common piece between both raw and downsampled segments. So, I pulled out a base class out of it.